### PR TITLE
Add watchOS and tvOS support

### LINKED ISSUE #234: Please update Starscream dependency up to '~> 3.1.1' how it was before in your `v2.7.0` release.
Hello,

In your previous releases `<= v2.7.0` the file: https://github.com/parse-community/ParseLiveQuery-iOS-OSX/blob/2.7.0/ParseLiveQuery.podspec

had the Starscream dependency like:

```
s.dependency 'Starscream', '~> 3.1.0'
```

but in your release `v2.7.1`, see: https://github.com/parse-community/ParseLiveQuery-iOS-OSX/pull/226

you guys changed it to:
```
s.dependency 'Starscream', '3.1.0'
```

NO `~>` in dependency line unfortunately.

Can you please revert back your changes to `s.dependency 'Starscream', '~> 3.1.0'` or better to `s.dependency 'Starscream', '~> 3.1.1` ?

Because otherwise, if I use another 3'rd party dependency in my project like [Cobrowse.io](https://github.com/cobrowseio/cobrowse-sdk-ios-binary) (which also requires Starscream) then I get pods conflicts error. In this case can't update ParseLiveQuery pod up to latest `v2.7.2`, because the following conflict error shown if I run `pod update`:

```
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "Starscream":
  In snapshot (Podfile.lock):
    Starscream (= 3.1.0)

  In Podfile:
    CobrowseIO/Extension was resolved to 2.8.8, which depends on
      Starscream (~> 3.1.1)

    ParseLiveQuery was resolved to 2.7.2, which depends on
      Starscream (= 3.1.0)
```

### PR DESCRIPTION
**Pull request by cbaker6**
- [x] Add watchOS and tvOS support
- [x] Add cache to travis and circle for faster builds
- [x] Move Carthage build to last in circle since it's the slowest and more important to see if iOS, macOS, and demo builds
- [x] Update project versions to prepare for release, updated the versions to 2.7.3
- [x] Update jazzy docs to newer version

This PR should probably be a release.

### CODE REVIEW

**Comment by cbaker6** [comment]
Note that Travis seems to be having some issues with Xcode builds, but I don't see the issue listed on travis. You may have to rerun the travis build if it fails to start

**Review by drdaz — APPROVED** [review]
Looks great.

**Review by drdaz — CHANGES_REQUESTED** [review]
Actually... the changelog needs an entry

**Review by drdaz — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,16 @@
 
 [Full Changelog](https://github.com/parse-community/ParseLiveQuery-iOS-OSX/compare/2.7.2...master)
```
Comment (TomWFox) [inline CHANGELOG.md]:
```suggestion
[Full Changelog](https://github.com/parse-community/ParseLiveQuery-iOS-OSX/compare/2.7.3...master)
```

**Review by TomWFox — CHANGES_REQUESTED** [review]

**Review by TomWFox — APPROVED** [review]

**Comment by cbaker6** [comment]
@TomWFox @drdaz can one of you push the pods? It looks like it failed to deploy and didn't give much info about the failure.

**Comment by cbaker6** [comment]
@TomWFox do you if the COCOAPODS_TRUNK_TOKEN is set? Travis us giving this “ [!] You need to run `pod trunk register` to register a session first.”

**Comment by TomWFox** [comment]
It is set, it has probably just expired - they do that periodically

**Comment by TomWFox** [comment]
Fairly sure the deployment is broken anyway because it work for the last release either

**Comment by cbaker6** [comment]
I tried a few different things in travis, but couldn't get it to go. I agree that the deployment seems broken.

I add this branch to push the pod manually for 2.7.3, https://github.com/parse-community/ParseLiveQuery-iOS-OSX/tree/2.7.3-pod

**Comment by cbaker6** [comment]
I found the problem...

During PR's to the master branch travis shows the following environment vars setup in the travis build log:
https://travis-ci.com/github/parse-community/ParseLiveQuery-iOS-OSX/jobs/441210418#L121
```
Setting environment variables from repository settings
120$ export GITHUB_TOKEN=[secure]
121$ export COCOAPODS_TRUNK_TOKEN=[secure]
```

I think the In the `COCOAPODS_TRUNK_TOKEN` isn't enabled for `tags` as it doesn't show in the logs anymore, the environment variable is missing. @TomWFox can you try to completely delete it and add it back enabling it for tags?
https://travis-ci.com/github/parse-community/ParseLiveQuery-iOS-OSX/jobs/441073847#L134
```
Setting environment variables from repository settings
134$ export GITHUB_TOKEN=[secure]
```

**Comment by cbaker6** [comment]
I think the `Only available to the master branch` is what's blocking it from tags. Setting it to `Available to all branches` should fix it

**Comment by TomWFox** [comment]
Sorry for the lack of response here, I've been having all sorts of issues with my shell since updating to Big Sur 🤦

@drdaz Can you get a new trunk token and put it in travis?

**Comment by drdaz** [comment]
I'll see if I can figure that out later today 😃

**Comment by drdaz** [comment]
I've updated the Cocoapods token in Travis and made it 'Available to all branches'.

**Comment by TomWFox** [comment]
Thanks!

**Comment by cbaker6** [comment]
I was able to deploy 2.7.3 and 2.8.0 via Travis by re-running the respective builds. The good thing about Travis is it can rerun the exact same build while pulling the latest env vars. We shouldn’t have an issue pushing via tags and Travis as long as the token doesn’t expire.

Also, the docs are deploying again via Travis and jazzy. Only thing is, the code in this repo is currently undocumented. I made the web link align with the iOS and Swift repos, http://parseplatform.org/ParseLiveQuery-iOS-OSX/api/

**Comment by TomWFox** [comment]
Awesome. The token expires after some period (a couple months roughly) if not used which is likely on this repo. I believe all that is needed is to run `pod trunk me` periodically which could be done with a [Cron job](https://docs.travis-ci.com/user/cron-jobs/).

**Comment by cbaker6** [comment]
I’ve seen cron in Travis, but I’ve never tried it. I wonder if setting the pod token to the same value as the iOS SDK (and eventually the Parse-Swift SDK) will fix the expiration problem since at least 1 of these should release within a 2 month period?

**Comment by TomWFox** [comment]
Thats definitely a good idea. I think it would still be reasonably conceivable for the token to expire but with the same token in each we could have one Cron job as an insurance.

**Pull request merged**